### PR TITLE
fix: could accept minify from user

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

Use the minify option when user provided and stop merging the default options. If not provided, use the default options.

## Checklist

- [x] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/config.ts [hunk 1]
```diff
@@ -247,6 +247,44 @@ export const composeAutoExternalConfig = (options: {
     : {};
 };
 
+export function composeMinifyConfig(
+  minify: NonNullable<RsbuildConfig['output']>['minify'],
+): RsbuildConfig {
+  if (minify !== undefined) {
+    return {
```
Comment (Timeless0911) [inline packages/core/src/config.ts:254]: Should we delete use's `output.minify` config before final config merge process. Or we can return `{}` here, and final merge process will ship user's minify config

Comment (fi3ework) [reply packages/core/src/config.ts]: Nice catch, I returned `{}` here.

File: packages/core/src/config.ts [hunk 2]
```diff
@@ -247,6 +247,41 @@ export const composeAutoExternalConfig = (options: {
...
+    // User's minify configuration will be merged afterwards.
+    return {};
+  }
+
+  // When minify is not specified, Rslib will use a sane default for minify options.
+  // The default options will only perform dead code elimination and unused code elimination.
+  return {
+    output: {
+      minify: {
+        js: true,
```
Comment (chenjiahan) [inline packages/core/src/config.ts:263]: Why do need minify JS files by default

Comment (chenjiahan) [reply packages/core/src/config.ts:263]: Libraries usually ship uon-minified code to npm.

Comment (fi3ework) [reply packages/core/src/config.ts:263]: By default, it's not minified. Adding `minify: true` is required as we want to perform some tree-shaking related compress options by default (https://github.com/web-infra-dev/rslib/blob/d68915dd2294a0a0d2e767e9ed672c25fdb2f6a8/packages/core/src/config.ts#L288-L294).

Comment (chenjiahan) [reply packages/core/src/config.ts:263]: get

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**